### PR TITLE
Local ci

### PR DESCRIPTION
**Pull request by knatten**
Instructions for how to run CI locally, and some improvements to make it work in Docker on Windows.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -102,3 +102,13 @@ Zivid Samples are distributed under the [BSD license](https://github.com/zivid/c
...
+
+## Development
+
+To run continuous integration locally, use [Docker](https://www.docker.com). With Docker installed, run this command:
+```
+docker run -it -v <unixy-repo-path>:/host -w /host/continuous-integration/linux ubuntu:18.04 bash -c "./setup.sh && ./lint.sh && ./build.sh"
+```
+Where <unixy-repo-path> is the unixy path to the repo on your computer. On Linux, use `$PWD` for this. On Windows you need to translate the windowsy path to a unixy one (e.g. `/c/Users/alice/Documents/cpp-extra-samples`).
+
+Running this takes a long time, since `setup.sh` takes several minutes to complete. Often it's useful to run CI locally many times, as you keep fixing your code. In that case, remove everything after `bash` in the above command. Then type `setup.sh` once to set it up, after which you can quickly run `./lint.sh && ./build.sh` repeatedly to check your code.
```
Comment (nedrebo) [inline README.md]: Maybe this shoult be the only recommended way. Then this Readme section can be simplified.

Comment (knatten) [reply README.md]: Good point! Fixed.

**Comment by knatten** [comment]
Ready for next round. I appreciate @nedrebo reviewing, and would also like someone on @SatjaSivcev's team to test it out.

**Comment by knatten** [comment]
@nedrebo did you intend to merge master into this branch? If not, I'll undo your merge and rebase on master.

**Comment by nedrebo** [comment]
> @nedrebo did you intend to merge master into this branch? If not, I'll undo your merge and rebase on master.

No, sorry about that.

**Comment by knatten** [comment]
Rebased on master now

**Comment by torbsorb** [comment]
@knatten, I'm running this now. I didn't have 'ubuntu:18.04' locally, so Docker pulled it from library/ubuntu (which I assume is good). Then I tried to run `./setup.sh`, but this fails on `bash: ./setup.sh: /bin/bash^M: bad interpreter: No such file or directory`. This seems to be a [carriage return Windows vs Linux issue](https://askubuntu.com/questions/304999/not-able-to-execute-a-sh-file-bin-bashm-bad-interpreter).

**Comment by nedrebo** [comment]
> @knatten, I'm running this now. I didn't have 'ubuntu:18.04' locally, so Docker pulled it from library/ubuntu (which I assume is good). Then I tried to run `./setup.sh`, but this fails on `bash: ./setup.sh: /bin/bash^M: bad interpreter: No such file or directory`. This seems to be a [carriage return Windows vs Linux issue](https://askubuntu.com/questions/304999/not-able-to-execute-a-sh-file-bin-bashm-bad-interpreter).

We have this in gcps:
```
$ cat .gitattributes 
* text=auto

# Linux files
*.sh text eol=lf
```

**Comment by knatten** [comment]
@torbsorb are you sure you are on the tip of the branch? @nedrebo I already fixed that in 863fa4e797b on Monday:
```
$ cat .gitattributes 
*.sh text eol=lf
```

**Comment by knatten** [comment]
Figured it out, he had a global git configuration which overrode the setting in `.gitattributes`. Works now.

**Comment by torbsorb** [comment]
I had to increase docker resources in order for G++ to work with Ninja. Maybe add a comment that Docker should be given "enough resources" (whatever that means). Maybe just add that "Ninja might struggle if Docker is not given enough resources. If you run into issues please try increasing the available resources"

**Comment by knatten** [comment]
Good suggestion! I incorporated that now while I squashed and rebased on master.

**Comment by SatjaSivcev** [comment]
Looking forward to trying this out.

**Pull request merged**